presto-hive/src/test/java/io/prestosql/plugin/hive/parquet/TestParquetDecimalScaling.java

```diff
@@ -143,6 +224,142 @@ public void testReadingMatchingPrecision(String tableName, boolean forceFixedLen
+                        38, 14,
+                        ImmutableList.of("10.0100", "10.0000", "1.2300", maximumValue(18, 4), minimumValue(18, 4))}
+        };
+    }
+
+    @DataProvider
+    public Object[][] roundableDecimals()
+    {
+        return new Object[][] {
+                {"decimal_10_2_round", false,
+                        10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        12, 1,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2))},
+                {"decimal_9_2_fle_round", true,
+                        9, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(9, 2), minimumValue(9, 2)),
+                        12, 1,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(9, 2), minimumValue(9, 2))},
+                {"decimal_4_2_round", false,
+                        4, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(4, 2), minimumValue(4, 2)),
+                        7, 1,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(4, 2), minimumValue(4, 2))},
+                {"decimal_10_2_round2", false,
+                        10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        12, 1,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2))}
+        };
+    }
+
+    @DataProvider
+    public Object[][] differentUnscalable()
+    {
+        // tableName, useFixedLengthArray, parquetPrecision, parquetScale, writeValues, schemaPrecision, schemaScale
+        return new Object[][] {
+                {"decimal_4_2_rescale_fail", false,
+                        4, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(4, 2), minimumValue(4, 2)),
+                        4, 3},
+                {"decimal_10_2_rescale_fail", false,
+                        10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        10, 3},
+                {"decimal_10_2_rescale_fail2", false,
+                        10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        3, 2},
+                {"decimal_10_2_fle_rescale_fail", true,
+                        10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        14, 7},
+                {"decimal_10_2_rescale_fail3", false,
+                        10, 2,
+                        ImmutableList.of("10.01", "10", "1.23", maximumValue(10, 2), minimumValue(10, 2)),
+                        10, 4},
```

Comment:
Why is this failure? is it because `maximumValue(10, 2)` doesn't fit? Then remove other test values, they are misleading. Maybe also use explicit value "99999999.99" instead if `maximumValue`, to make things more explicit.

Same for other negative test cases.